src/main/scala/org/apache/spark/shuffle/RssShuffleWriter.scala

```diff
@@ -201,8 +208,19 @@ class RssShuffleWriter[K, V, C](
 
   private def createDataBlock(buffer: Array[Byte]): ByteBuffer = {
     val uncompressedByteCount = buffer.size
-    val compressedBuffer = new Array[Byte](compressor.maxCompressedLength(uncompressedByteCount))
-    val compressedByteCount = compressor.compress(buffer, compressedBuffer)
+    var compressedBuffer: Array[Byte] = null
+    var compressedByteCount: Int = 0
+    if (Compression.COMPRESSION_CODEC_ZSTD.equals(compression)) {
+      compressedBuffer = new Array[Byte](uncompressedByteCount)
+      val n = Zstd.compress(compressedBuffer, buffer, 3)
```

Comment:
spark external shuffle service is using level 1. Would it be better to not hard code it, and make it an RssOpts instead?

Reply:
Good suggestion! Added conf `spark.shuffle.rss.compress.zstd.level`